src/macrofier/engine.mjs

```diff
@@ -1133,45 +1191,49 @@ function insertExampleMacros(template, examples, method, json, templates) {
   return template.replace(/\$\{method\.examples\}/g, content)
 }
 
-function generateResult(result, json, templates) {
-  const type = types.getSchemaType(result, json, { destination: state.destination, section: state.section  })
+function generateResult(schema, json, templates) {
+
+  let name = schema.name || schema.title
```

Comment:
Hmm... schemas never have a `name` attribute, only `title`. Are we adding the name to it somewhere else?

Ideally we'd pass it in as a parameter to this method, instead of mutating the schema object (which could cause side effects elsewhere in the code).

Reply:
updated signature with option name parameter and calling with proper name value